core/src/main/scala/org/locationtech/rasterframes/ref/RasterSource.scala

```diff
@@ -191,7 +191,25 @@ object RasterSource extends LazyLogging {
   /** A RasterFrames RasterSource which delegates most operations to a geotrellis-contrib RasterSource */
   abstract class DelegatingRasterSource(source: URI, delegateBuilder: () => GTRasterSource) extends RasterSource with URIRasterSource {
     @transient
-    lazy val delegate = delegateBuilder()
+    @volatile
+    private var _delRef: GTRasterSource = _
+
+    private def retryableRead[R >: Null](f: GTRasterSource => R): R = synchronized {
+      try {
+        if (_delRef == null)
+          _delRef = delegateBuilder()
+        f(_delRef)
+      }
+      catch {
+        // On this exeception we attempt to recreate the delegate and read again.
+        case be: java.nio.BufferUnderflowException =>
+          _delRef = null
+          val newDel = delegateBuilder()
+          val result = f(newDel)
+          _delRef = newDel
+          result
+      }
+    }
```

Comment:
This is the tricky bit. Need to make sure it's thread safe, even though multiple threads "shouldn't" be calling the read code.